### PR TITLE
Pmcmc dev

### PR DESCRIPTION
**Pull request by lwhittles**
I have added a diagnostic output to the pmcmc to help diagnose mixing issues, and corrected a bug which was the likely cause of said mixing issues!

### CODE REVIEW

File: tests/testthat/test-pmcmc.R [hunk 1]
```diff
@@ -75,7 +75,8 @@ test_that("pmcmc runs without error", {
                     dimnames = list(
                       c('beta', 'start_date'),
                       c('beta', 'start_date')
-                    ))
+                    )), 
+   output_proposals = TRUE
```
Comment (johnlees) [inline tests/testthat/test-pmcmc.R:79]: Add a quick test on the shape of output_proposals?

File: R/pmcmc.R [hunk 2]
```diff
@@ -347,6 +361,15 @@ pmcmc <- function(data,
                      'log_likelihood' = curr_ll, 
                      'log_posterior' = curr_lpost) 
     states[iter, ] <- curr_ss
+    
+    
+    if(output_proposals) {
+      proposals[iter, ] <- c(prop_pars,
```
[the remark this answers is not in the document]
Comment (lwhittles) [reply R/pmcmc.R:367]: Good idea, I'll take any speed-up going! I've removed the names from both the `res` and `proposals` matrices

File: tests/testthat/test-pmcmc.R (hunk 1 above)
Comment (lwhittles) [reply tests/testthat/test-pmcmc.R:79]: sorry, I misunderstood this! adding a check now!

Comment (lwhittles) [reply tests/testthat/test-pmcmc.R:79]: all sorted.

**Pull request merged**